test/FastTests/Utils/RevisionsHelper.cs

```diff
@@ -126,9 +127,28 @@ public static async Task SetupRevisionsAsync(IDocumentStore store, Raven.Server.
             var result = await store.Maintenance.SendAsync(new ConfigureRevisionsOperation(configuration));
 
             var documentDatabase = await serverStore.DatabasesLandlord.TryGetOrCreateResourceStore(store.Database);
+
             await documentDatabase.RachisLogIndexNotifications.WaitForIndexNotification(result.RaftCommandIndex.Value, serverStore.Engine.OperationTimeout);
         }
 
+        public static async Task SetupRevisionsOnShardedDatabaseAsync(IDocumentStore store, Raven.Server.ServerWide.ServerStore serverStore, RevisionsConfiguration configuration, IAsyncEnumerable<ShardedDocumentDatabase> shards)
+        {
+            if (store == null)
+                throw new ArgumentNullException(nameof(store));
+            if (serverStore == null)
+                throw new ArgumentNullException(nameof(serverStore));
+            if (configuration == null)
+                throw new ArgumentNullException(nameof(configuration));
+
+            var result = await store.Maintenance.SendAsync(new ConfigureRevisionsOperation(configuration));
+            var index = result.RaftCommandIndex;
+
+            await foreach (var shard in shards)
+            {
+                await shard.RachisLogIndexNotifications.WaitForIndexNotification(index.Value, serverStore.Engine.OperationTimeout);
+            }
+        }
```

Comment:
use `SetupRevisionsAsync` instead

Reply:
Cant do it because it doesn't wait for applying the config on the shards

Reply:
So we need to fix the EP instead

Reply:
It has no connection to any EP..
In the original methods you are waiting for notification directly on the 'documentDatabase'

Reply:
in 6.0 the EP are different and we should await on the EP level. please verify this.

Reply:
Done